### PR TITLE
add featuregate summarization for oc and api

### PR DESCRIPTION
**Pull request by deads2k**
adds analysis for serialized FeatureGates to produce table representations.

### CODE REVIEW

File: pkg/markdown/markdown.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+		body:                  &bytes.Buffer{},
+		orderedListItemNumber: make([]int, 10), // nine nested ordered lists should be enough for anyone
+	}
+}
+
+func (m *Markdown) Bytes() []byte {
+	ret := &bytes.Buffer{}
+	fmt.Fprintf(ret, "# %s\n\n", m.title)
+	fmt.Fprintf(ret, "## Table of Contents\n")
+	fmt.Fprintf(ret, m.tableOfContents.String())
```
Comment (bertinatto) [inline pkg/markdown/markdown.go:32]: nit: technically should be a format string (`%s`) or use `Fprint` instead

File: pkg/features/feature_diff_summary.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	for _, forClusterProfile := range i.featureInfo {
+		for featureSet := range forClusterProfile {
+			ret = append(ret, forClusterProfile[featureSet])
+		}
+	}
+	return ret
+}
+
+func (i *ReleaseFeatureDiffInfo) AllFeatureSets() sets.String {
+	allFeatureSets := sets.String{}
```
Comment (bertinatto) [inline pkg/features/feature_diff_summary.go:25]: This has been deprecated in favor of the generic version. There's a PR out to update library-go: https://github.com/openshift/library-go/pull/1643#pullrequestreview-1948942910

File: pkg/features/summarizer.go [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+	Disabled        sets.String
+	AllFeatureGates map[string]bool
+}
+
+type FilenameToContent map[string][]byte
+
+func (i *ReleaseFeatureInfo) CalculateDiff(ctx context.Context, from *ReleaseFeatureInfo) *ReleaseFeatureDiffInfo {
+	to := i
+	allFeatureSets := from.AllFeatureSets()
+	allFeatureSets.Insert(to.AllFeatureSets().List()...)
```
Comment (bertinatto) [inline pkg/features/summarizer.go:93]: `List()` doesn't seem to be implemented yet?

**Review by bertinatto — COMMENTED** [review]
Just a few random things I noticed...

**Comment by deads2k** [comment]
comments addressed.

**Pull request merged**